Implement simple IBD (#190)
planetarium/libplanet: CHANGES.md

 - Added `Swarm.DifferentVersionPeerEncountered` event handler that can handle
   events when a different version of a peer is discovered.  [[#167]], [[#185]]
 - Added `Peer.AppProtocolVersion` property.  [[#185]]

>  - `Swarm.StartAsync()` now receives the height(`Index`) of the block from

Reviewer: ```suggestion
 - `Swarm.StartAsync()` now receives the height of blocks (tip `Index`) from
```

   other known peers and synchronizes the blocks if necessary before starts

>    another process. [[#187]]

Reviewer: What does “another process” here mean?

Author: Receiving and processing messages(`Swarm.ReceiveMessageAsync()`) Repeating delta distribution(`Swarm.RepeatDeltaDistributionAsync()`), and so on.

Reviewer: It would be better if that's elaborated, e.g.:

> … before propagating/receiving pinpointed recent blocks to prevent inefficient round-trips.

 - Added `Swarm.DifferentVersionPeerEncountered` event handler that can handle
   events when a different version of a peer is discovered.  [[#167]], [[#185]]
 - Added `Peer.AppProtocolVersion` property.  [[#185]]
 - `Swarm.StartAsync()` now receives the height of blocks (tip `Index`) from

>    other known peers and synchronizes the blocks if necessary before starts

Reviewer: ```suggestion
   other known peers and synchronizes the blocks if necessary
```

   before propagating/receiving pinpointed recent blocks to prevent inefficient

>    round-trips [[#187]]

Reviewer: ```suggestion
   round-trips.  [[#187], [#190]]
```

[#185]: https://github.com/planetarium/libplanet/pull/185

> [#187]: https://github.com/planetarium/libplanet/issues/187

Reviewer: ```suggestion
[#187]: https://github.com/planetarium/libplanet/issues/187
[#190]: https://github.com/planetarium/libplanet/pull/190
```
